thirdeye-server/src/test/java/ai/startree/thirdeye/resources/AnomalyResourceTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package ai.startree.thirdeye.resources;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ai.startree.thirdeye.auth.AccessControlProvider;
+import ai.startree.thirdeye.auth.AuthorizationManager;
+import ai.startree.thirdeye.auth.ThirdEyePrincipal;
+import ai.startree.thirdeye.service.AnomalyService;
+import ai.startree.thirdeye.spi.api.AlertApi;
+import ai.startree.thirdeye.spi.api.AnomalyApi;
+import ai.startree.thirdeye.spi.api.AuthorizationConfigurationApi;
+import ai.startree.thirdeye.spi.datalayer.bao.AlertManager;
+import ai.startree.thirdeye.spi.datalayer.bao.AnomalyManager;
+import ai.startree.thirdeye.spi.datalayer.dto.AlertDTO;
+import ai.startree.thirdeye.spi.datalayer.dto.AnomalyDTO;
+import ai.startree.thirdeye.spi.datalayer.dto.AuthorizationConfigurationDTO;
+import java.util.List;
+import javax.ws.rs.BadRequestException;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.Test;
+
+public class AnomalyResourceTest {
+
+  private static AnomalyResource newAnomalyResource(
+      final AnomalyManager anomalyManager,
+      final AlertManager alertManager
+      ) {
+    return new AnomalyResource(new AnomalyService(
+        anomalyManager,
+        alertManager,
+        null,
+        new AuthorizationManager(
+            null,
+            AccessControlProvider.alwaysAllow,
+            alertManager,
+            null,
+            anomalyManager
+        )
+    ));
+  }
+
+  static ThirdEyePrincipal nobody() {
+    return new ThirdEyePrincipal("nobody", "");
+  }
+
+  @Test
+  void testGet() {
```

Comment:
thinking aloud.

by the look of it, this seems like a test for AnomalyService. It is setup more like an integration test than a unit test and the resource method is really a pass through in this case. thoughts?

Reply:
Yes, I agree, this is more of an integration test. I can convert it later.